web/src/jira/components/JiraIssueComboBox.tsx

```diff
@@ -0,0 +1,101 @@
+import { ListItem, ListItemText, Typography } from "@mui/material";
+import { ControllerProps, FieldValues, UseFormReturn } from "react-hook-form";
+import { FindDimensionOptionsDocument } from "../../graphql/generated/graphql";
+import { useQuery } from "@apollo/client";
+import useInfiniteScroll from "react-infinite-scroll-hook";
+import DimensionComboBox from "../../components/entry-dialog/DimensionComboBox";
+import { useDebounceValue } from "usehooks-ts";
+import { useEffect, useState } from "react";
+import { useJiraIssueOptions } from "./useJiraIssueOptions";
+
+type JiraIssueComboBoxProps<T extends FieldValues> = {
+  form: UseFormReturn<T>;
+  name: "issue";
+  title: string;
+  rules?: ControllerProps["rules"];
+};
+
+const JiraIssueComboBox = <T extends FieldValues>({
+  name,
+  ...params
+}: JiraIssueComboBoxProps<T>) => {
+  const { data, loading } = useQuery(FindDimensionOptionsDocument);
+
+  const nvKeys = data?.findDimensionOptions[name] || [];
+
+  const [searchFilter, setSearchFilter] = useDebounceValue("", 400);
+  const [debouncePending, setDebouncePending] = useState(false);
+
+  const { options, loadMore, hasNextPage, isFetching, error } = useJiraIssueOptions({
+    issueKeys: nvKeys,
+    searchFilter,
+    enabled: !loading,
+  });
+
+  const [sentryRef, { rootRef }] = useInfiniteScroll({
+    loading: isFetching,
+    hasNextPage: hasNextPage,
+    disabled: error,
+    onLoadMore: loadMore,
+    delayInMs: 0,
+    rootMargin: `0px 0px 1500px 0px`,
+  });
+
+  useEffect(() => {
+    setDebouncePending(false);
+  }, [searchFilter]);
+
+  if (error) {
+    return <DimensionComboBox form={params.form} name="issue" title={params.title} />;
```

Comment:
Might be a good idea to add a notification here as well. Warning level would probably be enough as an error here does not prevent the user from continuing. Errors here won't get logged to Kibana so this way we would at least have a chance to get user reports about errors.